app/src/shared/components/MobileNav/index.jsx

```diff
@@ -126,9 +126,9 @@ const MobileNav = compose(
                             <li>
                                 <Link
                                     className={styles.link}
-                                    to={routes.login({
-                                        redirect: location.pathname,
-                                    })}
+                                    to={routes.login(redirect !== '/' ? {
+                                        redirect,
+                                    } : {})}
```

Comment:
What does this do

Reply:
While being on the top page (root path), `Sign In` links in the top nav are not gonna have `redirect` param in their hrefs.